src/pages/single-video/SingleVideo.jsx

```diff
@@ -32,10 +32,16 @@ const SingleVideo = () => {
     removeFromWatchLater,
   } = useWatchLaterVideos();
 
+  const {
+    historyState: { history },
+    addToHistory,
+  } = useWatchHistory();
+
   const navigate = useNavigate();
   const video = videos.find(eachVideo => eachVideo._id === videoId);
   const isLiked = likedList.find(eachVideo => eachVideo._id === videoId);
   const isPresentInWatchLater = watchLaterVideos.find(eachVideo => eachVideo._id === videoId);
+  const isPresentInHistory = history.find(eachVideo => eachVideo?._id === videoId);
```

Comment:
u can abstract out line 43 and 44 by making a function which checks whether a video is in a playlist

Reply:
Done. Thanks for the review.